propolis/src/hw/chipset/i440fx.rs

```diff
@@ -178,10 +186,19 @@ impl I440Fx {
     }
 }
 impl Chipset for I440Fx {
-    fn pci_attach(&self, bdf: Bdf, dev: Arc<dyn pci::Endpoint>) {
-        assert!(bdf.bus.get() == 0);
-
-        self.pci_bus.attach(bdf, dev, Some(self.route_lintr(&bdf)));
+    fn pci_root_bus(&self) -> &pci::Bus {
+        &self.pci_bus
+    }
+    fn pci_router(&self) -> &Arc<pci::router::Router> {
+        &self.pci_router
+    }
+    fn pci_attach(
+        &self,
+        bus: &pci::Bus,
+        bdf: Bdf,
+        dev: Arc<dyn pci::Endpoint>,
+    ) {
+        bus.attach(bdf, dev, Some(self.route_lintr(&bdf)));
```

Comment:
The selection of a local interrupt pin should be contingent on this being bus 0 (and possibly `route_lintr` should assert that it is), since the bus/pin relationships are described via ACPI tables and the default tables we're using will only describe them for the root bus.